src/escapeExamples.cpp

```diff
@@ -0,0 +1,50 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// [[Rcpp::export]]
+std::string escapeExamples(std::string x) {
+  std::string out;
+  out.reserve(x.length() * 1.1);
+
+  char in_string = '\0';
+  bool in_escape = false;
+  bool in_comment = false;
+
+  std::string::const_iterator cur, end = x.end();
+  for (cur = x.begin(); cur != end; cur++) {
+    if (in_comment) { // inside comment
+      if (*cur == '\n') {
+        in_comment = false;
+      }
+    } else if (in_string == '\0') { // regular code
+      if (*cur == '#') {
+        in_comment = true;
+      } else if (*cur == '\'' || *cur == '"' || *cur == '`') {
+        in_string = *cur;
+      }
+    } else { // inside string/symbol
+      if (in_escape) {
+        in_escape = false;
+        if (*cur == 'l' || *cur == 'v') {
+          out += '\\';
+        } else if (*cur == '\\') {
+          out += "\\\\";
```

Comment:
So this will add two backslashes, with `*cur` it is three, which does not seem right. Why does this work?

Reply:
Never mind, got it finally. We wrote out the original escape already, and three more is four.

Reply:
Yeah, I don't understand it either but empirically it works.